src/middlewares/isAuthorUser.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-underscore-dangle */
+const isAuthorUser = (object, user, type) => {
+  switch (type) {
+    case 'user':
+      if (object._id === user._id) return true;
+      break;
+    case 'project':
+      if (object.owners?.includes(user._id)) return true;
+      break;
+    case 'comment':
+      if (
+        object.comments?.[
```

Comment:
the object here could be anything either project,user,comment and review thats why i am using switch to check the type when we send an object we should send a type also and based on that type we go the case related

Reply:
then why are you writing 'object.comments' if the object is already a comment here?

Reply:
because i think we cant get only comments because they are embedded and when we get comments in controller it will be already a project then comments inside that thats why i am sending the project here to filter on comments or reviews